File: kitty-data/internal/model/event.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type Event struct {
	ID         bson.ObjectID `bson:"_id,omitempty" json:"id"`
	EventType  string        `bson:"eventType" json:"eventType"`
	Operator   string        `bson:"operator" json:"operator"`
	Value      int64         `bson:"value" json:"value"`
	Timestamp  time.Time     `bson:"timestamp" json:"timestamp"`
	CreateTime time.Time     `bson:"createTime" json:"createTime"`
}

File: kitty-data-quarkus/src/main/java/icu/jiapeng/qk/model/Event.java
package icu.jiapeng.qk.model;

import io.quarkus.mongodb.panache.PanacheMongoEntity;
import io.quarkus.mongodb.panache.common.MongoEntity;
import org.bson.codecs.pojo.annotations.BsonProperty;

import java.time.Instant;

@MongoEntity(collection = "events_qk")
public class Event extends PanacheMongoEntity {

    @BsonProperty("eventType")
    public String eventType;

    @BsonProperty("operator")
    public String operator;

    @BsonProperty("value")
    public long value;

    @BsonProperty("detailJson")
    public String detailJson;

    @BsonProperty("timestamp")
    public Instant timestamp;

    @BsonProperty("createTime")
    public Instant createTime;
}
